pkg/trace/config/apply.go

```diff
@@ -466,7 +482,7 @@ func (c *AgentConfig) applyDatadogConfig() error {
 	if config.Datadog.GetBool("apm_config.internal_profiling.enabled") {
 		profilingSite := config.Datadog.GetString("internal_profiling.profile_dd_url")
 		if profilingSite == "" {
-			profilingSite = site
+			profilingSite = config.Datadog.GetString("site")
```

Comment:
This is all wrong. Not because of your PR, because of the documentation of this function. It should work like this, I think: https://github.com/DataDog/datadog-agent/blob/c09f2bc6d3a13e213bcfa4270c7c09fe5a7cceec/pkg/process/config/yaml_config.go#L246-L253